native/SampleApps/SmartSyncExplorer/SmartSyncExplorerCommon/SObjectDataManager.m

```diff
@@ -172,13 +172,13 @@ - (void)createLocalData:(SObjectData *)newData {
 - (void)updateLocalData:(SObjectData *)updatedData {
     [updatedData updateSoupForFieldName:kSyncManagerLocal fieldValue:@YES];
     [updatedData updateSoupForFieldName:kSyncManagerLocallyUpdated fieldValue:@YES];
-    [self.store upsertEntries:@[ updatedData.soupDict ] toSoup:[[updatedData class] dataSpec].soupName withExternalIdPath:kSObjectIdField error:nil];
+    [self.store upsertEntries:@[ updatedData.soupDict ] toSoup:[[updatedData class] dataSpec].soupName];
 }
 
 - (void)deleteLocalData:(SObjectData *)dataToDelete {
     [dataToDelete updateSoupForFieldName:kSyncManagerLocal fieldValue:@YES];
     [dataToDelete updateSoupForFieldName:kSyncManagerLocallyDeleted fieldValue:@YES];
-    [self.store upsertEntries:@[ dataToDelete.soupDict ] toSoup:[[dataToDelete class] dataSpec].soupName withExternalIdPath:kSObjectIdField error:nil];
```

Comment:
It was wrong to do a upsert withExternalIdPath:kSObjectIdField - when the record is locally created, that field will be empty, as a result a new record is inserted!